server/handler.go

```diff
@@ -103,6 +106,41 @@ func (h *Handler) ComQuery(
 
 	var r *sqltypes.Result
 	var proccesedAtLeastOneBatch bool
+
+	rowchan := make(chan sql.Row)
+	errchan := make(chan error)
+	quit := make(chan struct{})
+
+	// This goroutine will be select{}ed giving a change to Vitess to call the
+	// handler.CloseConnection callback and enforcing the timeout if configured
+	go func() {
+		for {
+			select {
+			case <-quit:
+				return
+			default:
+			}
+			row, err := rows.Next()
+			if err != nil {
+				errchan <- err
+				return
+			}
+			rowchan <- row
+		}
+	}()
+
+	// Default (big) waitTime is 3600, but it wont matter if there is no timeout
+	// because the loop will just iterate again. If there is a timeout, it will
+	// be enforced to ensure that Vitess has a chance to call Handler.CloseConnection()
+	totalTime := 0 * time.Second
+	waitTime := 10 * time.Second
```

Comment:
shouldn't this be 1 * time.Hour?

Reply:
It was a test with smaller default timeouts that somewhat remained, changed and fixed.